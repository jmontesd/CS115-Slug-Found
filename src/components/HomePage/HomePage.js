import React from 'react';
import './HomePage.scss';
import { compose } from 'redux';
import { firestoreConnect } from 'react-redux-firebase';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import logo from '../logo.jpg';
import SearchBox from '../SearchBox/SearchBox';
import Item from '../Item/Item';

export class HomePage extends React.Component {
  state = { searchItem: '' };

  handleInput = (e) => {
    console.log(e.target.value);
    this.setState({ searchItem: e.target.value });
  };

  render() {
    const { posts, isLoggedIn } = this.props;

    if (!isLoggedIn) return <Redirect to="/login" />;

    return (
      <div className="container" id="body">
        <img className="logo" src={logo} alt="Logo" width={300} inputMode="scale"  />
        <SearchBox handleInput={this.handleInput} />
        <div className="HomePage-container">
          {posts &&
            posts
              .slice()
              .sort((a, b) => b.createdAt - a.createdAt)
              .filter((post) => {
                return post.title.toLowerCase().includes(this.state.searchItem.toLowerCase());
              })
              .map((post) => <Item key={post.id} post={post} />)}
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  // get all posts from firebase
  const { posts } = state.firestore.ordered;
  return {
    posts,
    isLoggedIn: state.firebase.auth.uid,
  };
};
// export this component with the neccessary data
export default compose(
  connect(mapStateToProps),
  firestoreConnect([{ collection: 'posts' }]),
)(HomePage);
