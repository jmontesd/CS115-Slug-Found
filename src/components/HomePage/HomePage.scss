.HomePage-container {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  justify-content: space-between;
}
.sidenav {
  // Move sidebar down
  // top: 0;
  // margin: 64px 0px 0px 0px;
  position: fixed;
}
