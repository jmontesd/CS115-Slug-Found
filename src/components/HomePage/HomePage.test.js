import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { HomePage } from './HomePage';

// needed for shallow rendering
configure({ adapter: new Adapter() });

describe('<HomePage />', () => {
  it('should ', () => {
    const wrapper = shallow(<HomePage isLoggedIn />);
  });
});
