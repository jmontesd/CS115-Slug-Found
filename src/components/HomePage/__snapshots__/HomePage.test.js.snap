// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render HomePage properly 1`] = `ShallowWrapper {}`;
